datalad/distributed/create_sibling_ghlike.py

```diff
@@ -145,27 +145,30 @@ def __init__(self, url, credential, require_token=True, token_info=None):
 
         self.api_url = url
         self._user_info = None
+        self._set_request_headers(
+            credential,
+            f'An access token is required for {url}' \
+            + f'. {token_info}' if token_info else '',
+            require_token,
+        )
 
-        if credential is None:
-            credential = urlparse(url).netloc
-
-        token_instructions = f'An access token is required for {url}'
-        if token_info:
-            token_instructions += f'. {token_info}'
+    def _set_request_headers(self, credential_name, auth_info, require_token):
+        if credential_name is None:
+            credential_name = urlparse(self.api_url).netloc
 
         try:
-            self.auth = Token(credential)(
+            self.auth = Token(credential_name)(
                 instructions=token_instructions)['token']
```

Comment:
Isn't `token_instructions` undefined now?

Reply:
As in: Should be `auth_info`, right?

Reply:
Yes, correct. Interesting test setup that tolerates such problems.

Reply:
Amended accordingly.